src/main/kotlin/app/revanced/patcher/Context.kt

```diff
@@ -0,0 +1,116 @@
+package app.revanced.patcher
+
+import app.revanced.patcher.apk.Apk
+import app.revanced.patcher.apk.ApkBundle
+import app.revanced.patcher.apk.ResourceFile
+import app.revanced.patcher.util.method.MethodWalker
+import org.jf.dexlib2.iface.Method
+import org.w3c.dom.Document
+import java.io.Closeable
+import java.io.InputStream
+import java.io.OutputStream
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+
+/**
+ * A common interface to constrain [Context] to [BytecodeContext] and [ResourceContext].
+ * @param apkBundle the [ApkBundle] for this context.
+ */
+abstract class Context internal constructor(val apkBundle: ApkBundle)
+
+/**
+ * A context for the bytecode of an [Apk.Base] file.
+ *
+ * @param apkBundle The [ApkBundle] for this context.
+ */
+class BytecodeContext internal constructor(apkBundle: ApkBundle) : Context(apkBundle) {
+    /**
+     * The list of classes.
+     */
+    val classes = apkBundle.base.bytecodeData.classes
+
+    /**
+     * Create a [MethodWalker] instance for the current [BytecodeContext].
+     *
+     * @param startMethod The method to start at.
+     * @return A [MethodWalker] instance.
+     */
+    fun toMethodWalker(startMethod: Method): MethodWalker {
```

Comment:
`to` would suggest you're consuming `this`, which you aren't. How about `methodIterator`? (keeping Java's convention)
```suggestion
    fun methodIterator(startMethod: Method): MethodWalker {
```

Reply:
methodIterator is not conventional. Methods should be a verb not a name.

Reply:
Additionally it is not an iterator, so the semantic would be wrong in that regard as well. getMethodCallTracer would be a better example

Reply:
> methodIterator is not conventional.

It is, though? There's `iterator()`, but also `listIterator()`. Hence I went with `methodIterator()`.

> Additionally it is not an iterator, so the semantic would be wrong in that regard as well. getMethodCallTracer would be a better example

A walker ***is*** an iterator. Walking from 1 to 2 to 3 to 4 is the same as iterating from 1 to 2 to 3 to 4.
`getMethodCallTracer` sounds like an obnoxiously long name to me, and by Kotlin's convention, getters should not start with `get`. So `methodCallTracer()` or just `walkMethods()` would be better, in my opinion.

Reply:
> It is, though?

[No, methods should be verbs as explained](https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html#:~:text=Methods%20should%20be%20verbs%2C%20in,of%20each%20internal%20word%20capitalized.&text=Except%20for%20variables%2C%20all%20instance,with%20a%20lowercase%20first%20letter.)

Reply:
> methodIterator is not conventional

[No, it has to be a verb as explained earlier](https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html#:~:text=Methods%20should%20be%20verbs%2C%20in,of%20each%20internal%20word%20capitalized.&text=Except%20for%20variables%2C%20all%20instance,with%20a%20lowercase%20first%20letter).

Reply:
@oSumAtrIX How about the other things I mentioned?

Reply:
The convention already disqualifies the initial name, and for that reason also `methodCallTracer`. It is also not an iterator just because it traces method calls, so "walking" is not an iterator. 

> An Iterator is an object that can be used to loop through collections, like ArrayList and HashSet. It is called an "iterator" because "iterating" is the technical term for looping.

I don't like the term "walk". It is too far-fetched. What it does is trace calls. For that reason, it should be called `traceMethodCalls` or similar.

Reply:
Sure, let's go with that then.

Reply:
I renamed the function to `traceMethodCalls`